src/login/logind-button.c

```diff
@@ -255,30 +329,47 @@ static int button_dispatch(sd_event_source *s, int fd, uint32_t revents, void *u
                                            LOG_MESSAGE("Suspend key pressed short."),
                                            "MESSAGE_ID=" SD_MESSAGE_SUSPEND_KEY_STR);
-                                manager_handle_action(b->manager, INHIBIT_HANDLE_SUSPEND_KEY, b->manager->handle_suspend_key, b->manager->suspend_key_ignore_inhibited, true);
+                                manager_handle_action(b->manager, INHIBIT_HANDLE_SUSPEND_KEY, b->manager->handle_suspend_key, b->manager->suspend_key_ignore_inhibited, /* is_edge= */ true, b->seat);
                         }
                         break;
 
                 case KEY_SUSPEND:
                         if (b->manager->handle_hibernate_key_long_press != HANDLE_IGNORE && b->manager->handle_hibernate_key_long_press != b->manager->handle_hibernate_key) {
                                 log_debug("Hibernate key pressed. Further action depends on the key press duration.");
-                                start_long_press(b->manager, &b->manager->hibernate_key_long_press_event_source, long_press_of_hibernate_key_handler);
+                                start_long_press(b, &b->manager->hibernate_key_long_press_event_source, long_press_of_hibernate_key_handler);
                         } else {
                                 log_struct(LOG_INFO,
                                            LOG_MESSAGE("Hibernate key pressed short."),
                                            "MESSAGE_ID=" SD_MESSAGE_HIBERNATE_KEY_STR);
-                                manager_handle_action(b->manager, INHIBIT_HANDLE_HIBERNATE_KEY, b->manager->handle_hibernate_key, b->manager->hibernate_key_ignore_inhibited, true);
+                                manager_handle_action(b->manager, INHIBIT_HANDLE_HIBERNATE_KEY, b->manager->handle_hibernate_key, b->manager->hibernate_key_ignore_inhibited, /* is_edge= */ true, b->seat);
+                        }
+                        break;
+
+                case KEY_ESC:
+                        if (b->manager->handle_secure_attention_key != HANDLE_IGNORE) {
+                                if ((b->button_modifier_mask & (BUTTON_MODIFIER_LEFT_SHIFT | BUTTON_MODIFIER_RIGHT_SHIFT)) != 0 &&
+                                    (b->button_modifier_mask & (BUTTON_MODIFIER_LEFT_CTRL | BUTTON_MODIFIER_RIGHT_CTRL)) != 0 &&
+                                    (b->button_modifier_mask & (BUTTON_MODIFIER_LEFT_ALT | BUTTON_MODIFIER_RIGHT_ALT)) != 0) {
+                                        log_debug("Secure Attention Key sequence pressed short.");
+                                        manager_handle_action(b->manager, INHIBIT_SECURE_ATTENTION_KEY, b->manager->handle_secure_attention_key, /* ignore_inhibited= */ true, /* is_edge= */ true, b->seat);
+                                }
                         }
                         break;
+
+                default:
+                        if (keycode_modifier_mask != BUTTON_MODIFIER_NONE)
```

Comment:
this conditoin check is unnecessary. ORing the mask with zero has no effect, hence you can just drop this line